### PR TITLE
Allow Faraday 1.x

### PR DESCRIPTION
**Pull request by ykitamura-mdsol**
Added support for Faraday 1.x.

Main changes are [removing sub-class constants definition from `Faraday::Error`](https://github.com/lostisland/faraday/blob/master/UPGRADING.md#faraday-10) in:

- lib/mauth/client/remote_authenticator.rb
- lib/mauth/client/security_token_cacher.rb

```diff
- rescue ::Faraday::Error::ConnectionFailed, ::Faraday::Error::TimeoutError => e
+ rescue ::Faraday::ConnectionFailed, ::Faraday::TimeoutError => e
```

@mdsol/team-16 @mdsol/architecture-enablement

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,24 +2,33 @@ language: ruby
 cache: bundler
 
 rvm:
-  - 2.3.8
-  - 2.4.6
-  - 2.5.5
-  - 2.6.3
+  - 2.3
+  - 2.4
```
Comment (ykitamura-mdsol) [inline .travis.yml:6]: Do we still need to support `2.3` and `2.4`?

Comment (ykitamura-mdsol) [reply .travis.yml:6]: talked with @jfeltesse-mdsol and probably we should drop them with the next major release (`v7.0`)

**Review by jfeltesse-mdsol — COMMENTED** [review]
Sorry, somehow I thought I had submitted my review but it was there in limbo...

1 nits, otherwise LGTM!

File: .travis.yml [hunk 2]
```diff
@@ -2,24 +2,33 @@ language: ruby
...
+  - 2.5
+  - 2.6
+  - 2.7
+
+gemfile:
+  - gemfiles/faraday_0.x.gemfile
+  - gemfiles/faraday_1.x.gemfile
 
 before_install:
   - gem update --system -N > /dev/null && echo "Rubygems version $(gem --version)"
```
[the remark this answers is not in the document]
Comment (ykitamura-mdsol) [reply .travis.yml]: added in 8ee05cb

**Review by jfeltesse-mdsol — APPROVED** [review]

**Pull request merged**